test/libcommonserver/io/testio.cpp

```diff
@@ -134,8 +134,10 @@ void TestIo::testLogDirectoryPath() {
             ec = std::make_error_code(std::errc::not_enough_memory);
             return SyncPath{};
         });
-
+        auto logInstance = Log::_instance;
+        Log::_instance.reset();
         CPPUNIT_ASSERT(!_testObj->logDirectoryPath(logDirPath, ioError));
+        Log::_instance = logInstance;
```

Comment:
It would be nice if you could explain why this tweak is required.

Reply:
```
As IoHelper::logDirectoryPath() use the path returned by Log::instance()->getLogFilePath().parent_path() if Log::_instance is not null,
we need to reset Log::_instance to ensure that we are in the default state where the function generates the log directory path from the temp directory path.
```
I added it as a doc string. ([c0c683a](https://github.com/Infomaniak/desktop-kDrive/pull/159/commits/c0c683a96c206345afa604c7fa658a30a597abd9))